### PR TITLE
Added Scantrad presence

### PR DESCRIPTION
**Pull request by Kozou4ever**
Hey, I'm adding Scantrad presence, here are the proofs that the presence works perfectly:
- https://i.imgur.com/zDnHRWj.png
- https://i.imgur.com/E7gKtp2.png

### CODE REVIEW

File: websites/S/Scantrad/presence.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    if (document.location.pathname.includes("/")) {
+        
+        presenceData.details = "Viewing Page:";
+        presenceData.state = "Home page";
+        presenceData.startTimestamp = browsingStamp;
+        };
+    if (document.location.pathname.includes("/mangas")) {
+
+        let chapter = document.getElementById("selectCh") as HTMLSelectElement;
+        let page = document.getElementById("selectPg") as HTMLSelectElement;
```
Comment (EncryptedDev) [inline websites/S/Scantrad/presence.ts:22]: merge lets

File: websites/S/Scantrad/presence.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
+const presence = new Presence({
+    clientId: "817856303719907389"
+});
+
+const browsingStamp = Math.floor(Date.now() / 1000);
```
Comment (EncryptedDev) [inline websites/S/Scantrad/presence.ts:5]: merge consts

File: websites/S/Scantrad/presence.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+presence.on("UpdateData", async () => {
+    const presenceData: PresenceData = { 
+        largeImageKey: "logo",
+        startTimestamp: browsingStamp
+    };
+
+    if (document.location.pathname.includes("/")) {
+
```
Comment (EncryptedDev) [inline websites/S/Scantrad/presence.ts:14]: remove line

File: websites/S/Scantrad/dist/metadata.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    "id": "379362415428632576"
+  },
+  "service": "Scantrad",
+  "description": {
+    "en": "Scantrad is a free french website which allow you to read the latest scans of the moment.",
+    "fr": "Scantrad est un site internet français gratuit qui vous permet de lire les derniers scans du moment."
+  },
+  "url": [
+    "scantrad.net"
+  ],
```
Comment (EncryptedDev) [inline websites/S/Scantrad/dist/metadata.json:13]: youve made it an array, but it's only one link so make it a string

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
Hiya, look at my review & format your code with Prettier (your indentation is messed)

File: websites/S/Scantrad/dist/metadata.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    "name": "Kozou",
+    "id": "379362415428632576"
+  },
+  "service": "Scantrad",
+  "description": {
+    "en": "Scantrad is a free french website which allow you to read the latest scans of the moment.",
+    "fr": "Scantrad est un site internet français gratuit qui vous permet de lire les derniers scans du moment."
+  },
+  "url": "scantrad.net",
+
```
Comment (EncryptedDev) [inline websites/S/Scantrad/dist/metadata.json:12]: remove line

File: websites/S/Scantrad/dist/metadata.json [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    "id": "379362415428632576"
+  },
+  "service": "Scantrad",
+  "description": {
+    "en": "Scantrad is a free french website which allow you to read the latest scans of the moment.",
+    "fr": "Scantrad est un site internet français gratuit qui vous permet de lire les derniers scans du moment."
+  },
+  "url": "scantrad.net",
+  
+  "version": "1.0.1",
```
Comment (EncryptedDev) [inline websites/S/Scantrad/dist/metadata.json:13]: version should be 1.0.0 as it's still not published

File: websites/S/Scantrad/presence.ts [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        largeImageKey: "logo",
+        startTimestamp: browsingStamp
+    };
+    if (document.location.pathname.includes("/")) {
+        presenceData.details = "Viewing Page:";
+        presenceData.state = "Home page";
+        presenceData.startTimestamp = browsingStamp;
+        };
+    if (document.location.pathname.includes("/mangas")) {
+
```
Comment (EncryptedDev) [inline websites/S/Scantrad/presence.ts:17]: remove line

File: websites/S/Scantrad/presence.ts [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+        let chapter = document.getElementById("selectCh") as HTMLSelectElement,
+            page = document.getElementById("selectPg") as HTMLSelectElement;
+
+        presenceData.details = "Reading " + 
+            document.querySelector(".tl-titre").textContent + ":";
```
Comment (EncryptedDev) [inline websites/S/Scantrad/presence.ts:22]: fix indentation

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
more requests & **use prettier**

File: websites/S/Scantrad/presence.ts [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+    };
+    if (document.location.pathname.includes("/")) {
+        presenceData.details = "Viewing Page:";
+        presenceData.state = "Home page";
+        presenceData.startTimestamp = browsingStamp;
+        };
+    if (document.location.pathname.includes("/mangas")) {
+
+        let chapter = document.getElementById("selectCh") as HTMLSelectElement,
+            page = document.getElementById("selectPg") as HTMLSelectElement;
```
Comment (EncryptedDev) [inline websites/S/Scantrad/presence.ts:19]: What's the point of letting this when it's never reassigned?

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
and one more

File: websites/S/Scantrad/presence.ts (hunk 9 above)
Comment (Kozou4ever) [reply websites/S/Scantrad/presence.ts:19]: It's actually reassigned?

Comment (EncryptedDev) [reply websites/S/Scantrad/presence.ts:18]: it is not

Comment (EncryptedDev) [reply websites/S/Scantrad/presence.ts:18]: codacy agrees with me

**Review by ririxidev — CHANGES_REQUESTED** [review]
Heya @Kozou4ever!
You should check out thing(s) that I requested to change in my review. Please look at them!
- Fix DeepScan & Codacy.
- Support more subpages e.g. their forum.
- Please use ``else if`` statment too instead of only ``if`` bc it looks digusting and isn't really good thing :/

Regards,
**ririxi**. PreMiD Reviewer Team.
Contact: mail@ririxi.dev, ririxi#2721.

**Comment by ririxidev** [comment]
image

**Comment by Kozou4ever** [comment]
I litteraly wrote in my commit's description that it redirects to another website, do I have to create another presence for this forum?

**Review by StrikerFRFX — APPROVED** [review]
LGTM, the forum DOES redirect to an entirely different website which means if it were to be supported it could be under an entirely different presence. You could add support, but I personally do not see it necessary to jam it into this presence (especially due to it being a different domain entirely)

**Review by Bas950 — APPROVED** [review]
LGTM I guess...

File: websites/S/Scantrad/presence.ts [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+    if (document.location.pathname.includes("/mangas")) {
+        const chapter = document.getElementById("selectCh") as HTMLSelectElement,
+            page = document.getElementById("selectPg") as HTMLSelectElement;
+
+        presenceData.details = "Reading " + document.querySelector(".tl-titre").textContent + ":";
+        presenceData.state = chapter.options[chapter.selectedIndex].text.replace('Chapitre', 'Chapter') + " | " + page.options[page.selectedIndex].text;
+        presenceData.smallImageKey = "reading";
+        presenceData.smallImageText = "Reading a scan";
+        presenceData.startTimestamp = browsingStamp;
```
Comment (ririxidev) [inline websites/S/Scantrad/presence.ts:20]: Yeah uh why you add startTimestamp every time when you already have it added to presenceData object?

File: websites/S/Scantrad/presence.ts [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+        } 
+    else if (document.location.pathname.includes("/connexion") || document.location.pathname.includes("/inscription")) {
+        presenceData.details = "Viewing Page:";
+        presenceData.state = "Login page";
+        presenceData.startTimestamp = browsingStamp;
```
Comment (ririxidev) [inline websites/S/Scantrad/presence.ts:25]: same like above

File: websites/S/Scantrad/presence.ts [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+        }
+    else if (document.location.pathname.includes("/planning")) {
+        presenceData.details = "Viewing Page:";
+        presenceData.state = "Planning page";
+        presenceData.startTimestamp = browsingStamp;
```
Comment (ririxidev) [inline websites/S/Scantrad/presence.ts:30]: same like above

File: websites/S/Scantrad/presence.ts [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+        }
+    else if (document.location.pathname.includes("/rejoindre")) {
+        presenceData.details = "Viewing Page:";
+        presenceData.state = "Jobs page";
+        presenceData.startTimestamp = browsingStamp;
```
Comment (ririxidev) [inline websites/S/Scantrad/presence.ts:35]: same like above

File: websites/S/Scantrad/presence.ts [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+        }
+    else if (document.location.pathname.includes("/dons")) {
+        presenceData.details = "Viewing Page:";
+        presenceData.state = "Donation page";
+        presenceData.startTimestamp = browsingStamp;
```
Comment (ririxidev) [inline websites/S/Scantrad/presence.ts:40]: same like above

File: websites/S/Scantrad/presence.ts [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+        }
+    else {
+        presenceData.details = "Viewing Page:";
+        presenceData.state = "Home page";
+        presenceData.startTimestamp = browsingStamp;
```
Comment (ririxidev) [inline websites/S/Scantrad/presence.ts:45]: same like above

**Review by ririxidev — CHANGES_REQUESTED** [review]
k then, i have different thing.

File: websites/S/Scantrad/presence.ts (hunk 10 above)
Comment (Kozou4ever) [reply websites/S/Scantrad/presence.ts]: Fixed and done! :)

**Review by ririxidev — APPROVED** [review]
k

**Comment by Kozou4ever** [comment]
hey! so is it good or?

**Comment by SlowLife1661** [comment]
It's K just wait for another reviewer

**Review by StrikerFRFX — APPROVED** [review]
lgtm

**Pull request merged**